### PR TITLE
fix: properly flag shutdown is happening

### PR DESCRIPTION
**Pull request by pdecat**
Without this change, concurrent accesses to `PluginManager.shuttingDown()` will incorrectly report a shutdown is happening while there isn't.

Symptoms being, you see this in the logs:
```2024-07-30 15:01:14.075 UTC [WARN]  PluginManager Get failed for aws: plugin manager is shutting down (0xc009ba2450)```
but there is *no* corresponding `[INFO] PluginManager Shutdown`
https://github.com/turbot/steampipe/blob/12304d487387468a4768e2e8c70945b7f8cea82d/pkg/pluginmanager_service/plugin_manager.go#L257

### CODE REVIEW

File: pkg/pluginmanager_service/plugin_manager.go [hunk 1]
```diff
@@ -609,7 +611,7 @@ func (m *PluginManager) initializePlugin(connectionConfigs []*sdkproto.Connectio
 // return whether the plugin manager is shutting down
 func (m *PluginManager) shuttingDown() bool {
 	if !m.shutdownMut.TryLock() {
-		return true
+		return m.shutdown
```
Comment (pdecat) [inline pkg/pluginmanager_service/plugin_manager.go:614]: I admit this is not perfect as a `TryLock()` failure may happen between `m.shutdownMut.Lock()` and `m.shutdown = true` [above](https://github.com/turbot/steampipe/pull/4342/files#diff-808c81445373b3024f6ce0ffd6fc4f20a75e1aba01d6e18a6951a6324937685bR263).

If that is not desirable, maybe the non-blocking `TryLock()`  usage should be replaced by a blocking `Lock()` altogether.

**Comment by kaidaguerre** [comment]
Hey @pdecat sorry for delay looking at this (as ever) - I'm back from leave now so I'll take a look

File: pkg/pluginmanager_service/plugin_manager.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ type PluginManager struct {
 	// shutdown syncronozation
 	// do not start any plugins while shutting down
 	shutdownMut sync.Mutex
```
Comment (kaidaguerre) [inline pkg/pluginmanager_service/plugin_manager.go:58]: so I think this is a good approach - I think I'd do it more like this, using a RWmutex: 

```
shutdownMut sync.RWMutex
shuttingDown    bool

...



// return whether the plugin manager is shutting down
func (m *PluginManager) isShuttingDown() bool {
	m.shutdownMut.RLock()
	defer m.shutdownMut.RUnlock()
	return m.shuttingDown
}
 ...

func (m *PluginManager) Shutdown(*pb.ShutdownRequest) (resp *pb.ShutdownResponse, err error) {
	m.shutdownMut.Lock()
        m.shuttingDown = true
	m.shutdownMut.Unlock(
        ...
}
```

**Review by kaidaguerre — CHANGES_REQUESTED** [review]
I like the approach of adding a flag

see my comment below for suggestions

File: pkg/pluginmanager_service/plugin_manager.go (hunk 2 above)
Comment (pdecat) [reply pkg/pluginmanager_service/plugin_manager.go]: Applied your proposed changes!

**Pull request merged**